### ISSUE TITLE
Issue with Hyva theme : The element "topmenu_generic" can't have a child

### ISSUE THREAD

**Issue by gbobts** [SnowdogApps/magento2-menu#308]
Hello,
Are you aware of this issue ?
I believed that in file vendor/snowdog/module-menu/view/frontend/layout/default_hyva.xml
Code below should be modified to add aliases so instead of :
`        <referenceBlock name="topmenu_generic">
            <block name="topmenu.mobile" class="Snowdog\Menu\Block\Menu">
                <arguments>
                    <argument name="menu" xsi:type="string">hyva-topmenu-mobile</argument>
                </arguments>
            </block>
            <block name="topmenu.desktop" class="Snowdog\Menu\Block\Menu">
                <arguments>
                    <argument name="menu" xsi:type="string">hyva-topmenu-desktop</argument>
                </arguments>
            </block>
        </referenceBlock>`

We would have : 
`        <referenceBlock name="topmenu_generic">
            <block name="topmenu.mobile" as="topmenu.mobile.hyva" class="Snowdog\Menu\Block\Menu">
                <arguments>
                    <argument name="menu" xsi:type="string">hyva-topmenu-mobile</argument>
                </arguments>
            </block>
            <block name="topmenu.desktop" as="topmenu.desktop.hyva" class="Snowdog\Menu\Block\Menu">
                <arguments>
                    <argument name="menu" xsi:type="string">hyva-topmenu-desktop</argument>
                </arguments>
            </block>
        </referenceBlock>`

```
Error below (in exception.log, while browing any page of the site with Hyva theme) : 
main.CRITICAL: Magento\Framework\Exception\LocalizedException: The element "topmenu_generic" can't have a child because "topmenu_generic" already has a child with alias "topmenu.desktop". in vendor/magento/framework/Data/Structure.php:625
Stack trace:
#0 vendor/magento/framework/Data/Structure.php(329): Magento\Framework\Data\Structure->_insertChild()
#1 vendor/magento/framework/View/Layout/ScheduledStructure/Helper.php(193): Magento\Framework\Data\Structure->setAsChild()
#2 vendor/magento/framework/View/Layout/GeneratorPool.php(128): Magento\Framework\View\Layout\ScheduledStructure\Helper->scheduleElement()
#3 vendor/magento/framework/View/Layout/GeneratorPool.php(91): Magento\Framework\View\Layout\GeneratorPool->buildStructure()
#4 vendor/magento/framework/View/Layout.php(365): Magento\Framework\View\Layout\GeneratorPool->process()
#5 vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\Layout->generateElements()
#6 vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\Layout\Interceptor->___callParent()
#7 vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}()
#8 generated/code/Magento/Framework/View/Layout/Interceptor.php(32): Magento\Framework\View\Layout\Interceptor->___callPlugins()
#9 vendor/magento/framework/View/Layout/Builder.php(129): Magento\Framework\View\Layout\Interceptor->generateElements()
#10 vendor/magento/framework/View/Page/Builder.php(65): Magento\Framework\View\Layout\Builder->generateLayoutBlocks()
#11 vendor/magento/framework/View/Layout/Builder.php(65): Magento\Framework\View\Page\Builder->generateLayoutBlocks()
#12 vendor/magento/framework/View/Page/Config.php(227): Magento\Framework\View\Layout\Builder->build()
#13 vendor/magento/framework/View/Page/Config.php(591): Magento\Framework\View\Page\Config->build()
#14 vendor/magento/framework/View/Page/Config.php(549): Magento\Framework\View\Page\Config->getElementAttribute()
#15 vendor/magento/module-catalog/Helper/Product/View.php(218): Magento\Framework\View\Page\Config->addBodyClass()
#16 vendor/magento/module-catalog/Helper/Product/View.php(293): Magento\Catalog\Helper\Product\View->initProductLayout()
#17 vendor/magento/framework/Interception/Interceptor.php(58): Magento\Catalog\Helper\Product\View->prepareAndRender()
#18 vendor/magento/framework/Interception/Interceptor.php(138): Magento\Catalog\Helper\Product\View\Interceptor->___callParent()
#19 vendor/magento/framework/Interception/Interceptor.php(153): Magento\Catalog\Helper\Product\View\Interceptor->Magento\Framework\Interception\{closure}()
#20 generated/code/Magento/Catalog/Helper/Product/View/Interceptor.php(23): Magento\Catalog\Helper\Product\View\Interceptor->___callPlugins()
#21 vendor/magento/module-catalog/Controller/Product/View.php(178): Magento\Catalog\Helper\Product\View\Interceptor->prepareAndRender()
#22 vendor/magento/framework/Interception/Interceptor.php(58): Magento\Catalog\Controller\Product\View->execute()
#23 vendor/magento/framework/Interception/Interceptor.php(138): Magento\Catalog\Controller\Product\View\Interceptor->___callParent()
#24 vendor/magento/framework/Interception/Interceptor.php(153): Magento\Catalog\Controller\Product\View\Interceptor->Magento\Framework\Interception\{closure}()
#25 generated/code/Magento/Catalog/Controller/Product/View/Interceptor.php(23): Magento\Catalog\Controller\Product\View\Interceptor->___callPlugins()
#26 vendor/magento/framework/App/Action/Action.php(111): Magento\Catalog\Controller\Product\View\Interceptor->execute()
#27 vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\Action\Action->dispatch()
#28 vendor/magento/framework/Interception/Interceptor.php(138): Magento\Catalog\Controller\Product\View\Interceptor->___callParent()
#29 vendor/magento/framework/Interception/Interceptor.php(153): Magento\Catalog\Controller\Product\View\Interceptor->Magento\Framework\Interception\{closure}()
#30 generated/code/Magento/Catalog/Controller/Product/View/Interceptor.php(32): Magento\Catalog\Controller\Product\View\Interceptor->___callPlugins()
#31 vendor/magento/framework/App/FrontController.php(245): Magento\Catalog\Controller\Product\View\Interceptor->dispatch()
#32 vendor/magento/framework/App/FrontController.php(212): Magento\Framework\App\FrontController->getActionResponse()
#33 vendor/magento/framework/App/FrontController.php(146): Magento\Framework\App\FrontController->processRequest()
#34 vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\FrontController->dispatch()
#35 vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\FrontController\Interceptor->___callParent()
#36 vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php(99): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}()
#37 vendor/magento/framework/Interception/Interceptor.php(135): Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch()
#38 vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(71): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}()
#39 vendor/magento/framework/Interception/Interceptor.php(135): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch()
#40 vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}()
#41 generated/code/Magento/Framework/App/FrontController/Interceptor.php(23): Magento\Framework\App\FrontController\Interceptor->___callPlugins()
#42 vendor/magento/framework/App/Http.php(116): Magento\Framework\App\FrontController\Interceptor->dispatch()
#43 vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http->launch()
#44 pub/index.php(30): Magento\Framework\App\Bootstrap->run()
#45 {main} {"exception":"[object] (Magento\\Framework\\Exception\\LocalizedException(code: 0): The element \"topmenu_generic\" can't have a child because \"topmenu_generic\" already has a child with alias \"topmenu.desktop\". at vendor/magento/framework/Data/Structure.php:625)"} []
```

**Comment by jerrylopez** [comment]
I ran into this as well. I cleared out the snowdog layout by overriding it in my theme like this:

```
<!-- Path: app/design/frontend/Vendor/ThemeName/Snowdog_Menu/layout/override/base/default_hyva.xml -->

<?xml version="1.0"?>

<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <body>
        <!-- Clears out all Snowdog_Menu layout updates -->
    </body>
</page>
```

After overriding it, I just redefined it using whatever block name/alias I wanted.

**Comment by andrzejborkowski** [comment]
Hi,

thanks for reporting this issue, I've created an internal ticket DEV-104435 and we'll check it with our dev team soon.

**Comment by andrzejborkowski** [comment]
hi,

unfortunately, we were not able to check it yet. I'll update this issue once I get any update from our devs.

Thanks

**Comment by magento360** [comment]
We are also facing same issue 

Here is few videos about errors: https://www.loom.com/share/3a789da40a7842e3978f1c4a61c3cb32

**Comment by enriquetarazona** [comment]
Any update on this issue?

**Comment by sd2894** [comment]
Hi, 

[Create new menus](https://github.com/SnowdogApps/magento2-menu/wiki/Creating-a-menu) with identifier:
1. hyva-topmenu-desktop
2. hyva-topmenu-mobile
3. Hyva-menu-footer

![screenshot-divya m245 com-2024 04 15-10_22_30](https://github.com/SnowdogApps/magento2-menu/assets/43598410/11790269-39a2-4590-a3a1-490da40206ac)

app\code\Snowdog\Menu\view\frontend\layout\default_hyva.xml

 <block
            name="topmenu-desktop"
            class="Snowdog\Menu\Block\Menu"
        >
            <arguments>
                <argument name="menu" xsi:type="string">hyva-topmenu-desktop</argument>
            </arguments>
        </block>

**Comment by gbobts** [comment]
Yes, you can also upgrade the module as it was fixed by #316
